usr/share/rear/output/default/940_grub2_rescue.sh

```diff
@@ -158,17 +158,8 @@ if is_true $USING_UEFI_BOOTLOADER ; then
         echo "}"
     ) > $grub_config_dir/rear.cfg
 
-    # Tell rear.efi which configuration file to load
-    (   echo "search --no-floppy --fs-uuid --set=root $grub_boot_uuid"
-        echo ""
-        echo "set btrfs_relative_path=y"
-        echo "set prefix=(\$root)${grub_boot_dir}/grub${grub_num}"
-        echo ""
-        echo "configfile (\$root)${grub_boot_dir}/grub${grub_num}/rear.cfg"
-    ) > $grub_config_dir/rear_embed.cfg
-
     # Create rear.efi at UEFI default boot directory location.
-    build_bootx86_efi $boot_dir/efi/EFI/BOOT/rear.efi $grub_config_dir/rear_embed.cfg
+    build_bootx86_efi $boot_dir/efi/EFI/BOOT/rear.efi $grub_config_dir/rear.cfg "$boot_dir" /boot/efi
```

Comment:
@pcahyna why are you using `"$boot_dir" /boot/efi` function parameters for `rear.efi` but no function arguments `BOOTX64.efi`?

Reply:
First of all, `/boot/efi` is not really strictly needed here, but I felt it is convenient tp let the rescue Grub access the ESP, it would be a bit strange to have the partition from where we booted inaccessible.
Concerning `BOOTX64.efi`, we don't need to access this file anywhere and we don't even know that the regular Grub2 image is called `BOOTX64.efi`. By adding it, we would create an uneeded dependency on yet another hardcoded file path.

Reply:
I'd prefer to have all boot loaders created with same modules set, otherwise it might become pretty hard to troubleshoot booting issues.

V.

Reply:
I don't understand. The module set should be the same as in which case?

Reply:
I mean that `bootx64.efi` and `rear.efi` should contain same modules ...

Reply:
If you mean `BOOTX64.efi` created for the use on the ISO image in https://github.com/rear/rear/pull/2293/files#diff-2299bb436f1322dbe98ff15c1a26ba37R74 , this is a reasonable requirement. It could be easily done by adding the same arguments (`$boot_dir` and `/boot/efi`) to the second invocation.
(Now I see what you meant in your first comment, I misunderstood it, sorry.)

Reply:
@gozora, done. I also addressed your comment that `/boot/efi` should not be hardcoded. I am now using the UEFI_BOOTLOADER variable to determine the ESP, which is consistent with your change (https://github.com/gozora/rear/commit/b368b73a6b599f4d81bda45ff5b6a58547911c84) but independent on it.